src/main/java/com/sap/oss/phosphor/fosstars/data/github/BanditDataProvider.java

```diff
@@ -0,0 +1,187 @@
+package com.sap.oss.phosphor.fosstars.data.github;
+
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.RUNS_BANDIT_SCANS;
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.USES_BANDIT_SCAN_CHECKS;
+import static com.sap.oss.phosphor.fosstars.model.other.Utils.setOf;
+
+import com.sap.oss.phosphor.fosstars.model.Feature;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.ValueSet;
+import com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.value.ValueHashSet;
+import com.sap.oss.phosphor.fosstars.util.Yaml;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+import org.apache.commons.collections4.IteratorUtils;
+
+/**
+ * The data provider gathers info about how a project uses Bandit for static analysis. In
+ * particular, it tries to fill out the following features:
+ * <ul>
+ *   <li>{@link OssFeatures#RUNS_BANDIT_SCANS}</li>
+ *   <li>{@link OssFeatures#USES_BANDIT_SCAN_CHECKS}</li>
+ * </ul>
+ */
+public class BanditDataProvider extends GitHubCachingDataProvider {
+
+  /**
+   * A directory where GitHub Actions configs are stored.
+   */
+  private static final String GITHUB_ACTIONS_DIRECTORY = ".github/workflows";
+
+  /**
+   * A list of extensions of GitHub Actions configs.
+   */
+  private static final List<String> GITHUB_ACTIONS_CONFIG_EXTENSIONS
+      = Arrays.asList(".yaml", ".yml");
+
+  /**
+   * A step in a GitHub action that triggers analysis with CodeQL.
+   */
+  private static final Pattern RUN_STEP_BANDIT_REGEX_PATTERN
+      = Pattern.compile("^.*bandit .*$", Pattern.DOTALL);
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public BanditDataProvider(GitHubDataFetcher fetcher) {
+    super(fetcher);
+  }
+
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return setOf(RUNS_BANDIT_SCANS, USES_BANDIT_SCAN_CHECKS);
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
```

Comment:
Bandit can be run without GitHub actions. This data provider may be updated to catch it.

Reply:
#772 772